DuckDuckGo/Preferences/Model/AIChatPreferences.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  AIChatPreferences.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Combine
+import Foundation
+
+final class AIChatPreferences: ObservableObject {
+    static let shared = AIChatPreferences()
```

Comment:
💡 What do you think if we remove the singleton pattern? The one that acts as a singleton could be the `UserDefaults`, `AIChatPreferences` would listen to changes to those keys, and just emit the changes on user defaults.

Then, no matter how many instances of `AIChatPreferences` you create, the changes are going to be replicated anyway.

This is just a suggestion, not a blocker for this.

Reply:
Thought about it, but every single preference uses this pattern, I don't want to change a single one without changing others to introduce a different pattern to it, I believe consistency is better on this case.